File: app/src/main/java/com/islamic/Wazaker/ui/Fragments/EveningAzkarFragment/EveningAzkarAdapter.java
package com.islamic.Wazaker.ui.Fragments.EveningAzkarFragment;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.islamic.Wazaker.Data.Models.ZekerModel;
import com.islamic.wazkar.R;

import java.util.ArrayList;

import at.markushi.ui.CircleButton;

public class EveningAzkarAdapter extends RecyclerView.Adapter<EveningAzkarAdapter.EveningViewholder> {

    private final ArrayList<ZekerModel> models;

    public EveningAzkarAdapter(ArrayList<ZekerModel> nightItems) {
        this.models = nightItems;
    }

    @NonNull
    @Override
    public EveningViewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View v = layoutInflater.inflate(R.layout.morr_items, parent, false);

        return new EveningViewholder(v);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull EveningViewholder holder, @SuppressLint("RecyclerView") int position) {

        holder.zekertext.setText(models.get(position).getZeker());

        holder.copybtn.setOnClickListener(v -> {
            String zeker = models.get(position).getZeker();

            Intent sendIntent = new Intent();
            sendIntent.setAction(Intent.ACTION_SEND);
            sendIntent.putExtra(Intent.EXTRA_TEXT, zeker);
            sendIntent.setType("text/plain");

            Intent shareIntent = Intent.createChooser(sendIntent, null);
            v.getContext().startActivity(shareIntent);
        });


        final ZekerModel item = models.get(position);
        final int[] count = {item.getCurrentCount()};
        final Boolean[] checked = {item.getChecked()};
        holder.checkBox.setChecked(checked[0]);
        holder.plustext.setText(Integer.toString(count[0]));


        holder.plusbtn.setOnClickListener(v -> {
            count[0]++;

            item.setCurrentCount(count[0]);
            holder.plustext.setText(Integer.toString(count[0]));

            if (count[0] >= models.get(position).getCounter()) {
                checked[0] = true;
                item.setChecked(true);
                holder.checkBox.setChecked(checked[0]);
            }
        });
    }

    @Override
    public int getItemCount() {
        return models.size();
    }

    public static class EveningViewholder extends RecyclerView.ViewHolder {
        TextView zekertext, plustext;
        Button copybtn;
        CircleButton plusbtn;
        CheckBox checkBox;

        public EveningViewholder(@NonNull View itemView) {
            super(itemView);
            zekertext = itemView.findViewById(R.id.zekertext);
            copybtn = itemView.findViewById(R.id.copybtn);
            plusbtn = itemView.findViewById(R.id.pluss);
            plustext = itemView.findViewById(R.id.plustext);
            checkBox = itemView.findViewById(R.id.checkbox);
        }
    }
}

File: app/src/main/java/com/islamey/hamza/wazaker/ui/EveningAzkarFragment/EveningAzkarAdapter.kt
package com.islamey.hamza.wazaker.ui.EveningAzkarFragment


import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.AsyncListDiffer
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import com.islamey.hamza.wazaker.domain.Models.ZekerModel
import com.islamey.hamza.wazaker.utils.Utils.shareText
import com.islamey.wazkar.databinding.MorningItemBinding

class EveningAzkarAdapter : RecyclerView.Adapter<EveningAzkarAdapter.ViewHolder>() {

    inner class ViewHolder(val binding: MorningItemBinding) :
        RecyclerView.ViewHolder(binding.root) {
        init {
            binding.copybtn.setOnClickListener {
                shareText(itemView.context, differ.currentList[layoutPosition].zeker)
            }
        }

        fun bind(currentItem: ZekerModel) {
            var count = 0

            binding.plustext.text = count.toString()

            binding.plusButton.setOnClickListener {
                count++
                binding.plustext.text = count.toString()
                if (count >= currentItem.counter) {
                    removeItem(currentItem)
                }
            }
        }
    }

    private val diffCallback = object : DiffUtil.ItemCallback<ZekerModel>() {
        override fun areItemsTheSame(oldItem: ZekerModel, newItem: ZekerModel): Boolean {
            return oldItem.zeker == newItem.zeker
        }

        override fun areContentsTheSame(oldItem: ZekerModel, newItem: ZekerModel): Boolean {
            return oldItem == newItem
        }
    }

    val differ = AsyncListDiffer(this, diffCallback)

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): EveningAzkarAdapter.ViewHolder {
        val view = MorningItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(view)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: EveningAzkarAdapter.ViewHolder, position: Int) {

        val currentItem = differ.currentList[position]
        holder.bind(currentItem)

        holder.binding.zekertext.text = currentItem.zeker

    }

    private fun removeItem(currentItem: ZekerModel) {
        val newList = differ.currentList.toMutableList()
        newList.remove(currentItem)
        differ.submitList(newList)
    }

    override fun getItemCount(): Int {
        return differ.currentList.size
    }


}